test/node/tools/utils.js

```diff
@@ -218,3 +218,25 @@ module.exports.sorted = (iterable, how) => {
   items.sort(how);
   return items;
 };
+
+/** Deeply converts all ObjectIds into their hex value representation */
+const deepOidValue = obj => {
+  if (obj?._bsontype === 'ObjectId') {
+    return obj.toHexString();
+  }
+  if (Array.isArray(obj)) {
+    return obj.map(item => deepOidValue(item));
+  } else if (obj !== null && typeof obj === 'object' && !Buffer.isBuffer(obj)) {
+    return Object.entries(obj).reduce((acc, [key, value]) => {
+      acc[key] = deepOidValue(value);
+      return acc;
+    }, {});
+  }
+  return obj;
+};
+
+module.exports.assertDeepEqualsWithObjectId = (actual, expected) => {
```

Comment:
I appreciate this solution but I'll call out the source of truth for any two BSON values being equal is that they result in the same BSON bytes regardless of the properties and internals that they contain. I think it is important we keep the tests oriented towards what users can expect to rely upon in their code. So I was thinking that the places where we need to stop change usages of `expect(x).to.deep.equal(y)` could instead be: `expect(BSON.serialize(x)).to.deep.equal(BSON.serialize(y))`. Do you think we can adjust the tests to use (something like) this easily? 


Also, I apologize I didn't get back sooner about the team's consensus on deep equality. We discussed the pros and cons, and we feel as though this change is worth introducing despite the impact it has on the deep equality assumption. We already have BSON types that are not deep equal but serialize to the same BSON (Binary) and we cannot always avoid necessary private/internal changes to our classes. We're going to make sure the release notes clearly call out the change and identify a pretty easy workaround, setting the poolSize to 1! :)

```sh
$ node -p 'var BSON = require("."); util.isDeepStrictEqual(new BSON.ObjectId("0".repeat(24)), new BSON.ObjectId("0".repeat(24)))'
false
$ node -p 'var BSON = require("."); BSON.ObjectId.poolSize = 1; util.isDeepStrictEqual(new BSON.ObjectId("0".repeat(24)), new BSON.ObjectId("0".repeat(24)))'
true
```

Reply:
Did my best to update any failing tests to assert equality on the BSON bytes and I've removed that `assertDeepEqualsWithObjectId` implementation.